src/main/kotlin/org/malachite/estella/task/api/TaskController.kt

```diff
@@ -19,30 +20,41 @@ import org.springframework.web.bind.annotation.*
 @RequestMapping("/api/tasks")
 class TaskController(
         @Autowired private val taskService: TaskService,
+        @Autowired private val taskStageService: TaskStageService,
         @Autowired private val organizationService: OrganizationService,
 ) {
 
-    @Deprecated("Not tested yet - draft implementation, should be tested as part of ES-162")
     @CrossOrigin
     @Transactional
     @GetMapping
     fun getTasks(
-            @RequestParam("owner", required = false) organizationUuid: String?,
-            @RequestParam("taskStage", required = false) taskStageUuid: String?,
+            @RequestParam("owner") organizationUuid: String?,
+            @RequestParam("taskStage") taskStageUuid: String?,
+            @RequestParam("devMail") devMail: String?,
             @RequestHeader(EStellaHeaders.devPassword) password: String
     ): ResponseEntity<Any> {
-        if ((organizationUuid == null && taskStageUuid == null) || (organizationUuid != null && taskStageUuid != null))
-            return ResponseEntity.badRequest().body(Message("Exactly one of parameters: organizationUuid and taskStageUuid is required"))
+        if (!areParamsValid(organizationUuid, taskStageUuid, devMail))
+            return ResponseEntity.badRequest().body(Message("Exactly one of parameters: organizationUuid, taskStageUuid and devMail is required"))
         val tasks: List<TaskDto> = organizationUuid
                 ?.let {
-                    taskService.checkDevPassword(it, password)
-                            .getTasksByOrganizationUuid(it)
+                    devMail?.let { taskStageService.getTasksByDev(it, password) }
+                            ?:let{
```

Comment:
tutaj ten let jest niepotrzebny, a wgl całe to wyrażenie `?: ...` można by było wynieść poza tego aktualnego leta (ale to już taki przypisek redakcji)